src/adam_bashforth_adaptive.jl

```diff
@@ -0,0 +1,384 @@
+              errorNorm = 0.0
+              for i= 1:N
+                  sc[i] = abstol/N + max(abs(y[n+1][i]+diff[i]),abs(y[n+1][i]))*reltol
+              end
+              errorNorm=norm(truncation_error/sc)
+              normLE[2] = errorNorm/sqrt(N)
+
+              hratio = 0.0 #defined here to be accessible in both cases of if-statement
+              if normLE[2] <= 1 && maxstep > abs(h[n])
+                  #If the norm of truncation error is less than 1, and the step size
+                  #is smaller than the designated maximum step size, we can accept the
+                  #as is stated in Harier equation 7.4
+                  successful_step = true
+
+                  #If the step size h[n] is acceptable, it is time to set the next step size
+                  #Equation 7.5 in Harier should yield the next step size guess
+                  (h[n]*(1/normLE[2])^(1/(steporder+1))) #note: steporder +1  = k + 2
+
+                  #However, for good coding practice recommendation, the following careful
+                  #approach is taken to actually choose the step size so that there is
+                  #not too sudden a change in step size. Namely, it is allowed to increase
+                  #at most by a ratio of 'facmax', and at least by a ratio of 'facmin', which
+                  #are typically taken as 1.5 and 0.8. With in this range, the ratio between
+                  #h[n+1] and h[n] is choosen to suite the next step size guess, namely, hratio,
+                  #cushioned by a factor `fac`, typically choosen to be .8. Additionally, it most
+                  #be larger than the minstep.
+
+                  #All of these practices are written about in more detail on Harier pg 168
+
+                  ##So we set the desired ratio
+                  hratio= (1/normLE[2])^(1/(steporder+1)) #note: steporder = k + 1
+
+                  #Add then, following the conventions described above, we set the next step size
+                  h[n+1] = min(max(h[n]*max(min(facmax,fac*hratio),facmin),minstep),maxstep)
+
+                  #NOTE:To test with fixed step size, simply comment the above line, and
+                  #be sure to uncomment the line setting h = diff(tspan) on line 39
+              else
+                  #If the truncation error has norm > 1, then the step size is not acceptible
+                  #in this case, what we would take for the next step size, we use to reattempt
+                  #this n-th step
+                  successful_step = false
+                  h[n] = min(max(h[n]*max(min(facmax,fac*hratio),facmin),minstep),maxstep)
+
+                  #NOTE:To test with fixed step size, simply comment the above line, and
+                  #be sure to uncomment the line setting h = diff(tspan) on line 39
+              end
+
+          end #while loop over n-th step
+        end #end of loop for finding successfull step size
+
+        currentStepOrder = steporder
```

Comment:
shouldn't this be `current_steporder` to be consistent with `steporder`, `new_steporder`